### PR TITLE
Implement graceful shutdown from scratch.

### PR DESCRIPTION
**Pull request by ivanterekh**
Implemented suggestion for issue #9

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -16,6 +14,5 @@ func main() {
 	}
 
 	log.Printf("starting server on port %v", port) // TODO: change logging
-	server.Start(":" + port)
-	closer.Hold()
+	server.Run(":" + port)
```
Comment (dzyanis) [inline main.go]: I'd recommend to pass an address instead of a port

**Review by dzyanis — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (ivanterekh) [reply main.go]: Done.

File: main.go [hunk 2]
```diff
@@ -4,18 +4,15 @@ import (
...
-	"github.com/xlab/closer"
-
 	"github.com/ivanterekh/go-skeleton/server"
 )
 
 func main() {
-	port := os.Getenv("PORT")
-	if port == "" {
-		port = "8080"
+	address := os.Getenv("ADDRESS")
```
Comment (dzyanis) [inline main.go]: Should be encapsulated somehow. I'd recommend to create your own function. For example, you can create functions `env.GetString("NAME", "DEFAULT VALUE") string`, `env.GetInt("NAME", 42) int` and etc.

File: server/server.go [hunk 3]
```diff
@@ -4,30 +4,36 @@ import (
...
+			log.Fatal(err) //TODO: add logging
 		}
 	}()
+
+	quit := make(chan os.Signal)
+	signal.Notify(quit, os.Interrupt)
+	<-quit
+	log.Println("Shutdown Server ...") //TODO: add logging
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (dzyanis) [inline server/server.go]: there is some magic value ` 5*time.Second`. It should be a constant

**Review by dzyanis — CHANGES_REQUESTED** [review]

File: main.go [hunk 4]
```diff
@@ -2,20 +2,18 @@ package main
...
 	"log"
-	"os"
-
-	"github.com/xlab/closer"
 
+	"github.com/ivanterekh/go-skeleton/env"
 	"github.com/ivanterekh/go-skeleton/server"
 )
 
+type config struct {
```
Comment (dzyanis) [inline main.go]: I think we don't need this structure anymore

**Review by dzyanis — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (ivanterekh) [reply main.go]: Removed.

**Pull request merged**